src/courses/courses.service.ts

```diff
@@ -78,4 +79,44 @@ export class CourseService {
       );
     }
   }
+
+  /**
+   * It updates a course with the given id and body
+   * @param {string} id - string - the id of the course to be updated
+   * @param {UpdateCourseDto} body - UpdateCourseDto
+   * @returns The updated course
+   */
+  async update(id: string, body: UpdateCourseDto): Promise<Course> {
+    try {
+      const course = await this.coursesRepository.findOne(id);
+      if (!course) {
+        throw new BadRequestException(
+          ZuAppResponse.NotFoundRequest(
+            'Not Found',
+            'This course does not exist',
+          ),
+        );
+      }
+      if (body.rating) body.rating = parseFloat(body.rating.toFixed(1));
```

Comment:
why `toFixed(1)`?

Reply:
So we can have a single decimal value like 3.5